CLUtils/CogSLanguageUtilities.Core/Factories/Parser/ParserPoolManager.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Microsoft.CogSLanguageUtilities.Core.Services.Parser;
+using Microsoft.CogSLanguageUtilities.Definitions.APIs.Factories.Parser;
+using Microsoft.CogSLanguageUtilities.Definitions.APIs.Services;
+using Microsoft.CogSLanguageUtilities.Definitions.Configs.Consts;
+using Microsoft.CogSLanguageUtilities.Definitions.Exceptions.Parser;
+using Microsoft.CogSLanguageUtilities.Definitions.Models.Configs.Parser;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CogSLanguageUtilities.Core.Factories.Parser
+{
+    /*
+     * this is an object pool
+     * we don't initialize parsing service until it is requested
+     */
+    public class ParserPoolManager : IParserPoolManager
+    {
+        private ParserConfigModel _allParsersConfigs;
+        private string[] _validDocTypes;
+        private MSReadParserService _msreadParser;
+        private DocxParserService _docxParser;
+
+        public ParserPoolManager(ParserConfigModel allParsersConfigs)
+        {
+            _allParsersConfigs = allParsersConfigs;
+
+            // create valid types array
+            var list = new List<string>();
+            list.AddRange(Constants.MsReadValidFileTypes);
+            list.AddRange(Constants.DocxValidFileTypes);
+            _validDocTypes = list.ToArray();
```

Comment:
nit:
```csharp
_validDocTypes = new[] { Constants.MsReadValidFileTypes, Constants.DocxValidFileTypes };
```

Reply:
@moabba 
this expression is invalid! -> "can' initialize 1d array with other 1d arrays, it becomes 2d"
also re-checked how to concat two arrays, and still it's the same way we're doing it

Reply:
Sorry, didn't notice it was `AddRange`. Thought it was `Add`